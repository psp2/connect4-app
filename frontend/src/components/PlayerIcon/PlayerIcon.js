import React from "react";
import { Button } from "react-bootstrap";
import "./PlayerIcon.scss";

function PlayerIcon(props) {
  function forfeitGame(turn) {
    var base_url = 'http://127.0.0.1:5000/quit?id='
    var url = base_url.concat(props.gameId, '&turn=', turn)
    fetch(url, {method: 'put'})
    .then(response => response.json())
    .then(data => {
      console.log(data)
    })
    props.endGame(true);
  }

  function restart() {
    var base_url = 'http://127.0.0.1:5000/restart?id='
    var url = base_url.concat(props.gameId)
    fetch(url, {method: 'put'})
    .then(response => response.json())
    .then(data => {
      props.setGameState([data['turn'], data['state']])
      console.log(data)
    })
  }

  return (
    <div className="PlayerIcon">
      <div className="PlayerInfo">
        <div>Player {props.playerNumber}:</div>
        <div>{props.playerName}</div>
      </div>
      {props.playerNumber === 1 && (<div className="PlayerPicture1"></div>)}
      {props.playerNumber === 2 && (<div className="PlayerPicture2"></div>)}
      {props.playerNumber === 1 && (
        <div className="PlayerButtons">
          <Button className="Player1IconButton" onClick={props.undo}>UNDO</Button>
          <Button className="Player1IconButton" onClick={restart}>RESTART</Button>
          <Button onClick={() => forfeitGame(1)} className="Player1IconButton">
            FORFEIT
          </Button>
        </div>
      )}
      {props.playerNumber === 2 && (
        <div className="PlayerButtons">
          <Button className="Player2IconButton" onClick={props.undo}>UNDO</Button>
          <Button className="Player2IconButton" onClick={restart}>RESTART</Button>
          <Button onClick={() => forfeitGame(2)} className="Player2IconButton">
            FORFEIT
          </Button>
        </div>
      )}
    </div>
  );
}

export default PlayerIcon;
