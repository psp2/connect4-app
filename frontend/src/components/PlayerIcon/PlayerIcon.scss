.PlayerIcon {
  display: flex;
  flex-direction: column;
  width: 20vw;
  height: 73vh;
  color: white;
  font-family: "Roboto", sans-serif;
  font-size: 3em;
  margin: auto;
  border: solid white 1px;
  border-radius: 20px;
}

.PlayerInfo {
  display: flex;
  flex-direction: column;
  overflow: scroll;
}

.PlayerPicture1 {
  width: 16vw;
  height: 16vw;
  margin: auto;
  border: solid white 1px;
  border-radius: 20px;
  background-color: yellow;
}

.PlayerPicture2 {
  width: 16vw;
  height: 16vw;
  margin: auto;
  border: solid white 1px;
  border-radius: 20px;
  background-color: red;
}

.PlayerButtons {
  display: flex;
  flex-direction: column;
  margin: auto;
}

.Player1IconButton {
  margin: 1vh;
  width: 10vw;
  font-size: 0.5em;
  font-family: "Roboto", sans-serif;
  border-radius: 20px;
}

.Player1IconButton:hover {
  background-color: yellow;
  color: black;
  transition-duration: 0.3s;
}

.Player2IconButton {
  margin: 1vh;
  width: 10vw;
  font-size: 0.5em;
  font-family: "Roboto", sans-serif;
  border-radius: 20px;
}

.Player2IconButton:hover {
  background-color: red;
  color: white;
  transition-duration: 0.3s;
}
