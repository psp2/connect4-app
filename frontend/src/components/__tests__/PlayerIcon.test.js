import React from "react";
import PlayerIcon from "../PlayerIcon/PlayerIcon";
import renderer from "react-test-renderer";

it("playerIcon1, renders correctly", () => {
  const playerNumber = 1;
  const playerName = "Prashant";
  const tree = renderer
    .create(<PlayerIcon playerNumber={playerNumber} playerName={playerName} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("playerIcon2, renders correctly", () => {
  const playerNumber = 2;
  const playerName = "Vivek";
  const tree = renderer
    .create(<PlayerIcon playerNumber={playerNumber} playerName={playerName} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("playerIcon, null values", () => {
  const playerNumber = null;
  const playerName = null;
  const tree = renderer
    .create(<PlayerIcon playerNumber={playerNumber} playerName={playerName} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
