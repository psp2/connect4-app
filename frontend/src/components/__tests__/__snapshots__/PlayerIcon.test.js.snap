// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playerIcon, null values 1`] = `
<div
  className="PlayerIcon"
>
  <div
    className="PlayerInfo"
  >
    <div>
      Player 
      :
    </div>
    <div />
  </div>
</div>
`;

exports[`playerIcon1, renders correctly 1`] = `
<div
  className="PlayerIcon"
>
  <div
    className="PlayerInfo"
  >
    <div>
      Player 
      1
      :
    </div>
    <div>
      Prashant
    </div>
  </div>
  <div
    className="PlayerPicture1"
  />
  <div
    className="PlayerButtons"
  >
    <button
      className="Player1IconButton btn btn-primary"
      disabled={false}
      type="button"
    >
      UNDO
    </button>
    <button
      className="Player1IconButton btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      RESTART
    </button>
    <button
      className="Player1IconButton btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      FORFEIT
    </button>
  </div>
</div>
`;

exports[`playerIcon2, renders correctly 1`] = `
<div
  className="PlayerIcon"
>
  <div
    className="PlayerInfo"
  >
    <div>
      Player 
      2
      :
    </div>
    <div>
      Vivek
    </div>
  </div>
  <div
    className="PlayerPicture2"
  />
  <div
    className="PlayerButtons"
  >
    <button
      className="Player2IconButton btn btn-primary"
      disabled={false}
      type="button"
    >
      UNDO
    </button>
    <button
      className="Player2IconButton btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      RESTART
    </button>
    <button
      className="Player2IconButton btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      FORFEIT
    </button>
  </div>
</div>
`;
